src/clist.rs

```diff
@@ -165,21 +207,45 @@ impl<T> CList<T> {
         }
     }
 
-    #[inline]
+    /// Returns the number of elements in the list.
+    ///
+    /// # Returns 
+    ///
+    /// The number of elements in the list.
+   #[inline]
     pub fn len(&self) -> usize {
         self.len
     }
 
+    /// Checks if the list is empty
+    ///
+    /// # Returns 
+    ///
+    /// `true` if the list is empty, `false` otherwise.
     #[inline]
     pub fn is_empty(&self) -> bool {
         self.len == 0
     }
 
+    /// Asserts that the node at the given address is not free.
+    ///
+    /// # Arguments
+    ///
+    /// * `node_addr` - The address of the node.
     fn assert_not_free(&self, node_addr: &usize) {
         assert!(self.mem.not_free(node_addr), "Node address is free.");
     }
 
-    #[inline]
+    /// Gets a reference to the data at the given node address.
+    ///
+    /// # Arguments
+    ///
+    /// * `node_addr` - the address of the node.
+    ///
+    /// # Returns
+    ///
+    /// A reference to the data at the node address.
```

Comment:
It's important to mention that it panics if `ROOT_ADDR` is provided because the root node has no data